File: Python/reporte.py
from datetime import date

from estudiante import Estudiante


class Reporte:
    # El método depende del objeto Estudiante temporalmente para funcionar
    def generar_reporte_academico(self, estudiante: Estudiante):
        print("=== REPORTE GENERADO ===")
        print(f"Fecha: {date.today()}")
        print(f"Datos del alumno: {estudiante.get_nombre()}")
        print(f"Edad: {estudiante.get_edad()}")
        print("========================")

File: Java/Reporte.java
class Reporte {
    // El método depende del objeto Estudiante temporalmente para funcionar
    public void generarReporteAcademico(Estudiante estudiante) {
        System.out.println("=== REPORTE GENERADO ===");
        System.out.println("Fecha: " + java.time.LocalDate.now());
        System.out.println("Datos del alumno: " + estudiante.getNombre());
        System.out.println("Edad: " + estudiante.getEdad());
        System.out.println("========================");
    }
}